rasa/shared/core/flows/flow.py

```diff
@@ -1157,13 +1413,48 @@ def from_json(link_config: Any) -> FlowLink:
         """
         ...
 
+    def steps_in_tree(self) -> Generator[FlowStep, None, None]:
+        """Returns the steps in the tree of the flow link."""
+        ...
+
+    def child_steps(self) -> List[FlowStep]:
+        """Returns the child steps of the flow link."""
+        ...
+
 
 @dataclass
-class IfFlowLink:
+class BranchBasedLink:
+    target_reference: Union[Text, StepSequence]
+    """The id of the linked flow."""
+
+    def steps_in_tree(self) -> Generator[FlowStep, None, None]:
+        """Returns the steps in the tree of the flow link."""
+        if isinstance(self.target_reference, StepSequence):
+            yield from self.target_reference.steps
```

Comment:
I think if you don't do `steps_in_tree` for every step here, you'll only ever get one level deep from the main branch, but not the full tree.

Reply:
not quite i think: the target_reference is a stepsequence. a stepsequence doesn't have a `steps_in_tree` - as it arguably isn't a tree, but a list. it does have a `steps` though which includes all children, even if two levels deep:
```
    @property
    def steps(self) -> List[FlowStep]:
        """Returns the steps of the flow."""
        return [
            step
            for child_step in self.child_steps
            for step in child_step.steps_in_tree()
        ]
```